runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java

```diff
@@ -1624,7 +1630,15 @@ public TimerUpdateBuilder setTimer(TimerData setTimer) {
        */
       public TimerUpdateBuilder deletedTimer(TimerData deletedTimer) {
         deletedTimers.add(deletedTimer);
-        setTimers.remove(deletedTimer);
+        TimerData timerToDelete = null;
+        for (TimerData timer : setTimers) {
+          if (timer.getDomain().equals(deletedTimer.getDomain())
+              && timer.getNamespace().equals(deletedTimer.getNamespace())
+              && timer.getTimerId().equals(deletedTimer.getTimerId())) {
+            timerToDelete = timer;
+          }
+        }
+        setTimers.remove(timerToDelete);
```

Comment:
Should we warn the user if they try to clear an unset timer?

Reply:
Clearing an unset timer seems to be like a no-op to me.